test/unit_test/character_sheet_unit_tests/TalentTests.java

```diff
@@ -0,0 +1,170 @@
+package unit_test.character_sheet_unit_tests;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.BaseClasses;
+import rpg_database.character_sheet.Fields;
+import rpg_database.character_sheet.Focuses;
+import rpg_database.character_sheet.Talent;
+import rpg_database.character_sheet.TalentLevels;
+import rpg_database.character_sheet.Talents;
+import rpg_database.character_sheet.WeaponGroups;
+
+public class TalentTests {
+
+	// test setup
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	// test methods
+	@Test
+	public void testGetDefaultTalentLevel() {
+		Talent talent = new Talent(Talents.ALCHEMY);
+		assertEquals(TalentLevels.NOVICE, talent.getTalentLevel());
+	}
+
+	@Test
+	public void testGetMasterTalentLevel() {
+		Talent talent = new Talent(Talents.ALCHEMY, TalentLevels.MASTER);
+		assertEquals(TalentLevels.MASTER, talent.getTalentLevel());
+	}
+
+	@Test
+	public void testSetMasterTalentLevel() {
+		Talent talent = new Talent(Talents.ALCHEMY, TalentLevels.MASTER);
+		talent.setTalentLevel(TalentLevels.JOURNEYMAN);
+		assertEquals(TalentLevels.JOURNEYMAN, talent.getTalentLevel());
+	}
+
+	@Test
+	public void testGetTalentName() {
+		Talent talent = new Talent(Talents.ALCHEMY, TalentLevels.MASTER);
+		assertEquals(Talents.ALCHEMY, talent.getTalentName());
+	}
+
+	@Test
+	public void testGetAllowedBaseClassesAlchemy() {
+		Talent talent = new Talent(Talents.ALCHEMY);
+		HashSet<BaseClasses> expectedAllowedMageRogueWarrior = new HashSet<>(Arrays.asList(BaseClasses.MAGE, BaseClasses.ROGUE, BaseClasses.WARRIOR));
+		assertEquals(expectedAllowedMageRogueWarrior, talent.getAllowedBaseClasses());
+	}
+
+	@Test
+	public void testGetAllowedBaseClassesPoleWeaponStyle() {
+		Talent talent = new Talent(Talents.POLE_WEAPON_STYLE);
+		HashSet<BaseClasses> expectedAllowedBaseClassWarrior = new HashSet<>(Arrays.asList(BaseClasses.WARRIOR));
+		assertEquals(expectedAllowedBaseClassWarrior, talent.getAllowedBaseClasses());
+	}
+
+	@Test
+	public void testGetRequiredAttributeValuesAlchemy() {
+		Talent talent = new Talent(Talents.ALCHEMY);
+		HashMap<Fields, Integer> expectedAttributeRequirements = new HashMap<>();
+		expectedAttributeRequirements.put(Fields.CUNNING_VALUE, 3);
+		assertEquals(expectedAttributeRequirements, talent.getRequiredAttributeValues());
+	}
+
+	@Test
+	public void testGetRequiredAttributeValuesDualWeaponStyle() {
+		Talent talent = new Talent(Talents.DUAL_WEAPON_STYLE);
+		HashMap<Fields, Integer> expectedAttributeRequirements = new HashMap<>();
+		expectedAttributeRequirements.put(Fields.DEXTERITY_VALUE, 2);
+		assertEquals(expectedAttributeRequirements, talent.getRequiredAttributeValues());
+	}
+
```

Comment:
Kellene olyan teszt is amikor több attributum  az előfeltétele a specializáció felvételének.

Reply:
Nem értem hogy a talentekhez hogy jön a specializáció. Az nem scopeja ennek a tasknak. A több attribútumos viselkedés kezelése pedig a fieldRules szintjén van tesztelve. Nézd meg a getRequiredAttributes tesztjeit. Azt használja a Talent class. Itt csak azt tesztelem hogy ténylégesen azt is használja -e.

Btw. Az összes required attribute value csoportot le kell majd tesztelni parametrizált teszttel. Az a te feladatod lesz majd szóval ott ezt majd kiélheted ;)